Mathlib/Topology/UniformSpace/UniformConvergenceTopology.lean

```diff
@@ -652,6 +652,51 @@ protected theorem hasBasis_uniformity (h : 𝔖.Nonempty) (h' : DirectedOn (· 
   UniformOnFun.hasBasis_uniformity_of_basis α β 𝔖 h h' (𝓤 β).basis_sets
 #align uniform_on_fun.has_basis_uniformity UniformOnFun.hasBasis_uniformity
 
+variable {α β}
+
+/-- Let `t i` be a nonempty directed subfamily of `𝔖`
+such that every `s ∈ 𝔖` is included in some `t i`.
+Let `V` bounded by `p` be a basis of entourages of `β`.
+
+Then `UniformOnFun.gen 𝔖 (t i) (V j)` bounded by `p j` is a basis of entourages of `α →ᵤ[𝔖] β`. -/
+protected theorem hasBasis_uniformity_of_covering_of_basis {ι ι' : Type*} [Nonempty ι]
+    {t : ι → Set α} {p : ι' → Prop} {V : ι' → Set (β × β)} (ht : ∀ i, t i ∈ 𝔖)
+    (hdir : Directed (· ⊆ ·) t) (hex : ∀ s ∈ 𝔖, ∃ i, s ⊆ t i) (hb : HasBasis (𝓤 β) p V) :
+    (𝓤 (α →ᵤ[𝔖] β)).HasBasis (fun i : ι × ι' ↦ p i.2) fun i ↦
+      UniformOnFun.gen 𝔖 (t i.1) (V i.2) := by
+  have hne : 𝔖.Nonempty := (range_nonempty t).mono (range_subset_iff.2 ht)
+  have hd : DirectedOn (· ⊆ ·) 𝔖 := fun s₁ hs₁ s₂ hs₂ ↦ by
+    rcases hex s₁ hs₁, hex s₂ hs₂ with ⟨⟨i₁, his₁⟩, i₂, his₂⟩
+    rcases hdir i₁ i₂ with ⟨i, hi₁, hi₂⟩
+    exact ⟨t i, ht _, his₁.trans hi₁, his₂.trans hi₂⟩
+  refine (UniformOnFun.hasBasis_uniformity_of_basis α β 𝔖 hne hd hb).to_hasBasis
+    (fun ⟨s, i'⟩ ⟨hs, hi'⟩ ↦ ?_) fun ⟨i, i'⟩ hi' ↦ ⟨(t i, i'), ⟨ht i, hi'⟩, Subset.rfl⟩
+  rcases hex s hs with ⟨i, hi⟩
+  exact ⟨(i, i'), hi', UniformOnFun.gen_mono hi Subset.rfl⟩
+
+/-- If `t n` is a monotone sequence of sets in `𝔖`
+such that each `s ∈ 𝔖` is included in some `t n`
+and `V n` is an antitone basis of entourages of `β`,
+then `UniformOnFun.gen 𝔖 (t n) (V n)` is an antitone basis of entourages of `α →ᵤ[𝔖] β`. -/
+protected theorem hasAntitoneBasis_uniformity {t : ℕ → Set α} {V : ℕ → Set (β × β)}
```

Comment:
As I said above, I think we can just get rid of this lemma since the fact that the basis is antitone has nothing to do with countable generation. Do you have any other use for it?

Reply:
I found 3 lemmas that can be useful (i.e. that don't output new `HasAntitoneBasis` statements): [Filter.HasAntitoneBasis.tendsto](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/AtTopBot.html#Filter.HasAntitoneBasis.tendsto), [Filter.HasAntitoneBasis.eventually_subset](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/AtTopBot.html#Filter.HasAntitoneBasis.eventually_subset), and [Filter.HasAntitoneBasis.tendsto_smallSets](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/SmallSets.html#Filter.HasAntitoneBasis.tendsto_smallSets).

Reply:
Ah yes the diagonal basis is a good point, I should have looked more carefully. So let's keep that indeed. Could you replace `ℕ` by any index type satisfying [IsDirected](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Directed.html#IsDirected)?